src/Annotations/Command/QueueWatchCommand.php

```diff
@@ -66,8 +67,14 @@ protected function process(InputInterface $input, QueueItem $item, $entity = nul
                 $sanitized_display_name = $display_name;
             }
             $user = new User($id, $email, $sanitized_display_name);
-            $upsert = $this->hypothesisSdk->users()->upsert($user)->wait();
-            $this->logger->info(sprintf('Hypothesis user "%s" successfully %s.', $upsert->getUsername(), ($upsert->isNew() ? 'created' : 'updated')));
+            try {
+                $upsert = $this->hypothesisSdk->users()->upsert($user)->wait();
+                $this->logger->info(sprintf('Hypothesis user "%s" successfully %s.', $upsert->getUsername(), ($upsert->isNew() ? 'created' : 'updated')));
+            } catch (BadResponse $e) {
+                // If upsert fails, log error but don't repeat.
```

Comment:
Probably only want this on certain status codes (eg 400s)?

Reply:
I have added a check in the `catch` statement targetting responses `400` and `404`. Any other response and it will reattempt.

Reply:
I'd suggest not retrying any `4xx` response. It's a client error, so I wouldn't expect a retry to work.